connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -427,4 +435,62 @@ public void ensureValid(String name, Object value) {
         }
     }
 
+    private static class ResponseHttpHeadersValidator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            String strValue = (String) value;
+            if (strValue.isEmpty()) {
+                return;
+            }
+
+            String[] configs = StringUtil.csvSplit(strValue);
+            Arrays.stream(configs).forEach(config -> validateHttpResponseHeaderConfig(config));
+        }
+
+        private static void validateHttpResponseHeaderConfig(String config) {
+            try {
+                // validate format
+                String[] configTokens = config.trim().split("\\s+", 2);
+                if (configTokens.length != 2) {
+                    throw new ConfigException(String.format("Invalid format of header config \"%s\". "
+                            + "Expected: \"[ation] [header name]:[header value]\"", config));
+                }
+
+                // validate action
+                String method = configTokens[0].trim();
+                validateHeaderConfigAction(method);
+
+                // validate header name and header value pair
+                String header = configTokens[1];
+                String[] headerTokens = header.trim().split(":");
+                if (headerTokens.length != 2) {
+                    throw new ConfigException(
+                            String.format("Invalid format of header name and header value pair \"%s\". "
+                                    + "Expected: \"[header name]:[header value]\"", header));
+                }
+
+                // validate header name
+                String headerName = headerTokens[0].trim();
+                if (headerName.isEmpty() || headerName.contains(" ")) {
```

Comment:
Shouldn't this look for other whitespace characters, per the exception message? Something like:
```suggestion
                if (headerName.isEmpty() || headerName.matches("\\s")) {
```